MonoGame.Framework/Graphics/Vertices/VertexBuffer.cs

```diff
@@ -54,13 +54,18 @@ internal protected override void GraphicsDeviceResetting()
 
         public void GetData<T> (int offsetInBytes, T[] data, int startIndex, int elementCount, int vertexStride) where T : struct
         {
+            var elementSizeInBytes = Marshal.SizeOf(typeof(T));
+
+            if (vertexStride == 0)
+                vertexStride = elementSizeInBytes;

```

Comment:
Yes - XNA appears to follow this same behaviour of treating `0` as `sizeof(T)`. This case is specifically covered in the unit tests, because it surprised me too.